Header header
string id_check
bool safe
string desc
string extra